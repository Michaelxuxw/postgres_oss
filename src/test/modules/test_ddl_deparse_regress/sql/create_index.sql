-- Currently, ddl deparse only supports the TABLE command. Therefore, this
-- test file is only prepared for testing the TABLE command.

-- Command prepared for alter_table.sql.
-- For "ADD table_constraint_using_index"
CREATE TABLE test_add_constraint_using_index(
    id1 int,
    id2 int,
    id3 int,
    id4 int,
    id5 int,
    id6 int,
    id7 int
);
CREATE UNIQUE INDEX test_add_constraint_used_index1 ON test_add_constraint_using_index (id1);
CREATE UNIQUE INDEX test_add_constraint_used_index2 ON test_add_constraint_using_index (id2);
CREATE UNIQUE INDEX test_add_constraint_used_index3 ON test_add_constraint_using_index (id3);
CREATE UNIQUE INDEX test_add_constraint_used_index4 ON test_add_constraint_using_index (id4);
CREATE UNIQUE INDEX test_add_constraint_used_index5 ON test_add_constraint_using_index (id5);
CREATE UNIQUE INDEX test_add_constraint_used_index6 ON test_add_constraint_using_index (id6);
CREATE UNIQUE INDEX test_add_constraint_used_index7 ON test_add_constraint_using_index (id7);
