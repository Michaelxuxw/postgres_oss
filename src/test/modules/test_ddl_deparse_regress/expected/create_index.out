---
--- CREATE_INDEX
---
-- Command prepared for alter_table.sql.
-- For "ADD table_constraint_using_index"
CREATE TABLE test_add_constraint_using_index(
    id1 int,
    id2 int,
    id3 int,
    id4 int,
    id5 int,
    id6 int,
    id7 int
);
NOTICE:  deparsed json: {"fmt": "CREATE %{persistence}s TABLE %{if_not_exists}s %{identity}D (%{table_elements:, }s) %{inherits}s %{partition_by}s %{access_method}s %{with_clause}s %{on_commit}s %{tablespace}s", "identity": {"objname": "test_add_constraint_using_index", "schemaname": "public"}, "inherits": {"fmt": "INHERITS (%{parents:, }D)", "parents": null, "present": false}, "on_commit": {"fmt": "ON COMMIT %{on_commit_value}s", "present": false, "on_commit_value": null}, "tablespace": {"fmt": "TABLESPACE %{tablespace}I", "present": false, "tablespace": null}, "persistence": "", "with_clause": {"fmt": "WITH", "present": false}, "partition_by": {"fmt": "PARTITION BY %{definition}s", "present": false, "definition": null}, "access_method": {"fmt": "USING %{access_method}I", "present": false, "access_method": null}, "if_not_exists": "", "table_elements": [{"fmt": "%{name}I %{coltype}T STORAGE %{colstorage}s %{compression}s %{collation}s %{not_null}s %{default}s %{generated_column}s", "name": "id1", "type": "column", "coltype": {"typmod": "", "typarray": false, "typename": "int4", "schemaname": "pg_catalog"}, "default": {"fmt": "DEFAULT", "present": false}, "not_null": "", "collation": {"fmt": "COLLATE %{name}D", "name": null, "present": false}, "colstorage": "PLAIN", "compression": {"fmt": "COMPRESSION %{compression_method}I", "present": false, "compression_method": null}, "generated_column": {"fmt": "GENERATED ALWAYS AS", "present": false}}, {"fmt": "%{name}I %{coltype}T STORAGE %{colstorage}s %{compression}s %{collation}s %{not_null}s %{default}s %{generated_column}s", "name": "id2", "type": "column", "coltype": {"typmod": "", "typarray": false, "typename": "int4", "schemaname": "pg_catalog"}, "default": {"fmt": "DEFAULT", "present": false}, "not_null": "", "collation": {"fmt": "COLLATE %{name}D", "name": null, "present": false}, "colstorage": "PLAIN", "compression": {"fmt": "COMPRESSION %{compression_method}I", "present": false, "compression_method": null}, "generated_column": {"fmt": "GENERATED ALWAYS AS", "present": false}}, {"fmt": "%{name}I %{coltype}T STORAGE %{colstorage}s %{compression}s %{collation}s %{not_null}s %{default}s %{generated_column}s", "name": "id3", "type": "column", "coltype": {"typmod": "", "typarray": false, "typename": "int4", "schemaname": "pg_catalog"}, "default": {"fmt": "DEFAULT", "present": false}, "not_null": "", "collation": {"fmt": "COLLATE %{name}D", "name": null, "present": false}, "colstorage": "PLAIN", "compression": {"fmt": "COMPRESSION %{compression_method}I", "present": false, "compression_method": null}, "generated_column": {"fmt": "GENERATED ALWAYS AS", "present": false}}, {"fmt": "%{name}I %{coltype}T STORAGE %{colstorage}s %{compression}s %{collation}s %{not_null}s %{default}s %{generated_column}s", "name": "id4", "type": "column", "coltype": {"typmod": "", "typarray": false, "typename": "int4", "schemaname": "pg_catalog"}, "default": {"fmt": "DEFAULT", "present": false}, "not_null": "", "collation": {"fmt": "COLLATE %{name}D", "name": null, "present": false}, "colstorage": "PLAIN", "compression": {"fmt": "COMPRESSION %{compression_method}I", "present": false, "compression_method": null}, "generated_column": {"fmt": "GENERATED ALWAYS AS", "present": false}}, {"fmt": "%{name}I %{coltype}T STORAGE %{colstorage}s %{compression}s %{collation}s %{not_null}s %{default}s %{generated_column}s", "name": "id5", "type": "column", "coltype": {"typmod": "", "typarray": false, "typename": "int4", "schemaname": "pg_catalog"}, "default": {"fmt": "DEFAULT", "present": false}, "not_null": "", "collation": {"fmt": "COLLATE %{name}D", "name": null, "present": false}, "colstorage": "PLAIN", "compression": {"fmt": "COMPRESSION %{compression_method}I", "present": false, "compression_method": null}, "generated_column": {"fmt": "GENERATED ALWAYS AS", "present": false}}, {"fmt": "%{name}I %{coltype}T STORAGE %{colstorage}s %{compression}s %{collation}s %{not_null}s %{default}s %{generated_column}s", "name": "id6", "type": "column", "coltype": {"typmod": "", "typarray": false, "typename": "int4", "schemaname": "pg_catalog"}, "default": {"fmt": "DEFAULT", "present": false}, "not_null": "", "collation": {"fmt": "COLLATE %{name}D", "name": null, "present": false}, "colstorage": "PLAIN", "compression": {"fmt": "COMPRESSION %{compression_method}I", "present": false, "compression_method": null}, "generated_column": {"fmt": "GENERATED ALWAYS AS", "present": false}}, {"fmt": "%{name}I %{coltype}T STORAGE %{colstorage}s %{compression}s %{collation}s %{not_null}s %{default}s %{generated_column}s", "name": "id7", "type": "column", "coltype": {"typmod": "", "typarray": false, "typename": "int4", "schemaname": "pg_catalog"}, "default": {"fmt": "DEFAULT", "present": false}, "not_null": "", "collation": {"fmt": "COLLATE %{name}D", "name": null, "present": false}, "colstorage": "PLAIN", "compression": {"fmt": "COMPRESSION %{compression_method}I", "present": false, "compression_method": null}, "generated_column": {"fmt": "GENERATED ALWAYS AS", "present": false}}]}
NOTICE:  re-formed command: CREATE  TABLE  public.test_add_constraint_using_index (id1 pg_catalog.int4 STORAGE PLAIN     , id2 pg_catalog.int4 STORAGE PLAIN     , id3 pg_catalog.int4 STORAGE PLAIN     , id4 pg_catalog.int4 STORAGE PLAIN     , id5 pg_catalog.int4 STORAGE PLAIN     , id6 pg_catalog.int4 STORAGE PLAIN     , id7 pg_catalog.int4 STORAGE PLAIN     )      
CREATE UNIQUE INDEX test_add_constraint_used_index1 ON test_add_constraint_using_index (id1);
NOTICE:  deparsed json: {"fmt": "CREATE %{unique}s INDEX %{concurrently}s %{if_not_exists}s %{name}I ON %{only}s %{table}D USING %{index_am}s %{definition}s NULLS DISTINCT %{with}s %{tablespace}s %{where_clause}s", "name": "test_add_constraint_used_index1", "only": "", "with": {"fmt": "WITH", "present": false}, "table": {"objname": "test_add_constraint_using_index", "schemaname": "public"}, "unique": "UNIQUE", "index_am": "btree", "definition": "(id1 pg_catalog.int4_ops)", "tablespace": {"fmt": "TABLESPACE", "present": false}, "concurrently": "", "where_clause": {"fmt": "WHERE", "present": false}, "if_not_exists": ""}
NOTICE:  re-formed command: CREATE UNIQUE INDEX   test_add_constraint_used_index1 ON  public.test_add_constraint_using_index USING btree (id1 pg_catalog.int4_ops) NULLS DISTINCT   
CREATE UNIQUE INDEX test_add_constraint_used_index2 ON test_add_constraint_using_index (id2);
NOTICE:  deparsed json: {"fmt": "CREATE %{unique}s INDEX %{concurrently}s %{if_not_exists}s %{name}I ON %{only}s %{table}D USING %{index_am}s %{definition}s NULLS DISTINCT %{with}s %{tablespace}s %{where_clause}s", "name": "test_add_constraint_used_index2", "only": "", "with": {"fmt": "WITH", "present": false}, "table": {"objname": "test_add_constraint_using_index", "schemaname": "public"}, "unique": "UNIQUE", "index_am": "btree", "definition": "(id2 pg_catalog.int4_ops)", "tablespace": {"fmt": "TABLESPACE", "present": false}, "concurrently": "", "where_clause": {"fmt": "WHERE", "present": false}, "if_not_exists": ""}
NOTICE:  re-formed command: CREATE UNIQUE INDEX   test_add_constraint_used_index2 ON  public.test_add_constraint_using_index USING btree (id2 pg_catalog.int4_ops) NULLS DISTINCT   
CREATE UNIQUE INDEX test_add_constraint_used_index3 ON test_add_constraint_using_index (id3);
NOTICE:  deparsed json: {"fmt": "CREATE %{unique}s INDEX %{concurrently}s %{if_not_exists}s %{name}I ON %{only}s %{table}D USING %{index_am}s %{definition}s NULLS DISTINCT %{with}s %{tablespace}s %{where_clause}s", "name": "test_add_constraint_used_index3", "only": "", "with": {"fmt": "WITH", "present": false}, "table": {"objname": "test_add_constraint_using_index", "schemaname": "public"}, "unique": "UNIQUE", "index_am": "btree", "definition": "(id3 pg_catalog.int4_ops)", "tablespace": {"fmt": "TABLESPACE", "present": false}, "concurrently": "", "where_clause": {"fmt": "WHERE", "present": false}, "if_not_exists": ""}
NOTICE:  re-formed command: CREATE UNIQUE INDEX   test_add_constraint_used_index3 ON  public.test_add_constraint_using_index USING btree (id3 pg_catalog.int4_ops) NULLS DISTINCT   
CREATE UNIQUE INDEX test_add_constraint_used_index4 ON test_add_constraint_using_index (id4);
NOTICE:  deparsed json: {"fmt": "CREATE %{unique}s INDEX %{concurrently}s %{if_not_exists}s %{name}I ON %{only}s %{table}D USING %{index_am}s %{definition}s NULLS DISTINCT %{with}s %{tablespace}s %{where_clause}s", "name": "test_add_constraint_used_index4", "only": "", "with": {"fmt": "WITH", "present": false}, "table": {"objname": "test_add_constraint_using_index", "schemaname": "public"}, "unique": "UNIQUE", "index_am": "btree", "definition": "(id4 pg_catalog.int4_ops)", "tablespace": {"fmt": "TABLESPACE", "present": false}, "concurrently": "", "where_clause": {"fmt": "WHERE", "present": false}, "if_not_exists": ""}
NOTICE:  re-formed command: CREATE UNIQUE INDEX   test_add_constraint_used_index4 ON  public.test_add_constraint_using_index USING btree (id4 pg_catalog.int4_ops) NULLS DISTINCT   
CREATE UNIQUE INDEX test_add_constraint_used_index5 ON test_add_constraint_using_index (id5);
NOTICE:  deparsed json: {"fmt": "CREATE %{unique}s INDEX %{concurrently}s %{if_not_exists}s %{name}I ON %{only}s %{table}D USING %{index_am}s %{definition}s NULLS DISTINCT %{with}s %{tablespace}s %{where_clause}s", "name": "test_add_constraint_used_index5", "only": "", "with": {"fmt": "WITH", "present": false}, "table": {"objname": "test_add_constraint_using_index", "schemaname": "public"}, "unique": "UNIQUE", "index_am": "btree", "definition": "(id5 pg_catalog.int4_ops)", "tablespace": {"fmt": "TABLESPACE", "present": false}, "concurrently": "", "where_clause": {"fmt": "WHERE", "present": false}, "if_not_exists": ""}
NOTICE:  re-formed command: CREATE UNIQUE INDEX   test_add_constraint_used_index5 ON  public.test_add_constraint_using_index USING btree (id5 pg_catalog.int4_ops) NULLS DISTINCT   
CREATE UNIQUE INDEX test_add_constraint_used_index6 ON test_add_constraint_using_index (id6);
NOTICE:  deparsed json: {"fmt": "CREATE %{unique}s INDEX %{concurrently}s %{if_not_exists}s %{name}I ON %{only}s %{table}D USING %{index_am}s %{definition}s NULLS DISTINCT %{with}s %{tablespace}s %{where_clause}s", "name": "test_add_constraint_used_index6", "only": "", "with": {"fmt": "WITH", "present": false}, "table": {"objname": "test_add_constraint_using_index", "schemaname": "public"}, "unique": "UNIQUE", "index_am": "btree", "definition": "(id6 pg_catalog.int4_ops)", "tablespace": {"fmt": "TABLESPACE", "present": false}, "concurrently": "", "where_clause": {"fmt": "WHERE", "present": false}, "if_not_exists": ""}
NOTICE:  re-formed command: CREATE UNIQUE INDEX   test_add_constraint_used_index6 ON  public.test_add_constraint_using_index USING btree (id6 pg_catalog.int4_ops) NULLS DISTINCT   
CREATE UNIQUE INDEX test_add_constraint_used_index7 ON test_add_constraint_using_index (id7);
NOTICE:  deparsed json: {"fmt": "CREATE %{unique}s INDEX %{concurrently}s %{if_not_exists}s %{name}I ON %{only}s %{table}D USING %{index_am}s %{definition}s NULLS DISTINCT %{with}s %{tablespace}s %{where_clause}s", "name": "test_add_constraint_used_index7", "only": "", "with": {"fmt": "WITH", "present": false}, "table": {"objname": "test_add_constraint_using_index", "schemaname": "public"}, "unique": "UNIQUE", "index_am": "btree", "definition": "(id7 pg_catalog.int4_ops)", "tablespace": {"fmt": "TABLESPACE", "present": false}, "concurrently": "", "where_clause": {"fmt": "WHERE", "present": false}, "if_not_exists": ""}
NOTICE:  re-formed command: CREATE UNIQUE INDEX   test_add_constraint_used_index7 ON  public.test_add_constraint_using_index USING btree (id7 pg_catalog.int4_ops) NULLS DISTINCT   
